File: 2011/20110817 - dependencias - javascript/dependencias.js
function dependencias(diretas) {
    var transitivas = diretas;
    for(chave in diretas){
        diretas[chave].forEach(function (direta) {
            if (!diretas[direta]) {
                transitivas[direta] = [];
            } else {
                transitivas[chave] = transitivas[chave].concat(diretas[direta]);
            }
        });
        
    };
    return transitivas;
}

File: 2015/20150506 - dependencias - ruby/dependencias.rb
def dependencias(dicionario)
  result = dicionario.dup

  result.keys.each do |chave|
    result.values.each do |values|
      if values.include?(chave)
        values.concat(dicionario[chave])
      end
    end
  end

  result.values.each do |values|
    values.uniq!
    values.sort!
  end

  result
end